platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -400,19 +417,19 @@ define(
          * pre-calculate optimal column sizes without having to render
          * every row.
          */
-        MCTTableController.prototype.findLargestRow = function (rows) {
+        MCTTableController.prototype.buildLargestRow = function (rows) {
             var largestRow = rows.reduce(function (largestRow, row) {
                 Object.keys(row).forEach(function (key) {
-                    var currentColumn = row[key].text,
+                    var currentColumn = (row[key] || {}).text,
```

Comment:
If a column is missing, the proper behavior would be to skip the column and move on, ex:

```javascript
if (!row[key]) {
  // do nothing because a nil column can't be the largest.
} else if (!largestRow[key]) { 
  // row[key] is by default the largest we've seen so set it and move on.
  largestRow[key] = JSON.parse(JSON.stringify(row[key]));
} else {
  // compare lengths and set accordingly
}
return largestRow;
```

We don't want to allocate throw-away objects when we should be taking a different code path-- sure sometimes that helps deal with unknown inputs, but doing that inside of a loop that potentially contains tens of thousands of iterations could be quite wasteful.